packages/truffle-debugger/test/transaction.js

```diff
@@ -0,0 +1,98 @@
+import debugModule from "debug";
+const debug = debugModule("test:transaction");
+
+import { assert } from "chai";
+
+import Ganache from "ganache-cli";
+import Web3 from "web3";
+
+import { prepareContracts } from "./helpers";
+import Debugger from "lib/debugger";
+
+import sessionSelector from "lib/session/selectors";
+
+
+
+const __FAILURE = `
+pragma solidity ^0.4.24;
+
+contract FailureTest {
+  function run() {
+    revert();
+  }
+}
+`;
+
+const __SUCCESS = `
+pragma solidity ^0.4.24;
+
+contract SuccessTest {
+  function run() {
+    var x = 0;
+  }
+}
+`;
+
+let sources = {
+  "FailureTest.sol": __FAILURE,
+  "SuccessTest.sol": __SUCCESS,
+};
+
+describe("Transactions", function () {
+  var provider;
+  var web3;
+
+  var abstractions;
+  var artifacts;
+
+  before("Create Provider", async function() {
+    provider = Ganache.provider({seed: "debugger", gasLimit: 7000000});
+    web3 = new Web3(provider);
+  });
+
+  before("Prepare contracts and artifacts", async function() {
+    this.timeout(30000);
+
+    let prepared = await prepareContracts(provider, sources);
+    abstractions = prepared.abstractions;
+    artifacts = prepared.artifacts;
+  });
+
+  it("correctly marks a failed transaction as failed", async function() {
+    let instance = await abstractions.FailureTest.deployed();
+    //HACK: because this transaction fails, we have to extract the hash from
+    //the resulting exception (there is supposed to be a non-hacky way but it
+    //does not presently work)
+    let txHash;
+    try {
+      await instance.run(); //this will throw because of the revert
+    }
+    catch(error) {
+      txHash = error.hashes[0]; //it's the only hash involved
+    }
+
+    let bugger = await Debugger.forTx(txHash, {
+      provider,
+      contracts: artifacts
+    });
+
+    let session = bugger.connect();
+
+    assert.ok(!session.view(sessionSelector.transaction.receipt).status);
+  });
+
+  it("correctly marks a successful transaction as successful", async function() {
+    let instance = await abstractions.SuccessTest.deployed();
+    let receipt = await instance.run();
+    let txHash = receipt.tx;
+
+    let bugger = await Debugger.forTx(txHash, {
+      provider,
+      contracts: artifacts
+    });
+
+    let session = bugger.connect();
+
+    assert.ok(session.view(sessionSelector.transaction.receipt).status);
+  });
+});
```

Comment:
It might be good to test the new behavior here, namely that you can inspect variables after the trace finishes (in either success or failure case)

Reply:
Hm, yeah, that could be worth doing.  So, a third test, then...

Reply:
You could also just add the asserts to the existing two tests. Check that the status is correct _and_ check that the variables are what you expect them to be.

Reply:
A thought: Should the file be renamed then?  I named it `transaction.js` on the idea that, hey, this is where the tests related to transactiong-saving go.  Whereas this is rather different.

Reply:
I guess I'll call it `endstate.js`.

Reply:
Yeah in fact I think I'm going to make this a separate test, because it's actually rather different -- the two tests so far *don't* depend on advancing the debugger to the end.  (It took me a while to realize why this new test wasn't working!)  Better to make it separate.